pkg/collector/python/datadog_agent.go

```diff
@@ -185,3 +186,20 @@ func ReadPersistentCache(key *C.char) *C.char {
 	}
 	return TrackedCString(data)
 }
+
+var obfuscator = obfuscate.NewObfuscator(nil)
+
+// ObfuscateSQL obfuscates & normalizes the provided SQL query, writing the error into errResult if the operation
+// fails
+//export ObfuscateSQL
+func ObfuscateSQL(rawQuery *C.char, errResult **C.char) *C.char {
+	s := C.GoString(rawQuery)
+	obfuscatedQuery, err := obfuscator.ObfuscateSQLString(s)
+	if err != nil {
+		// memory will be freed by caller
+		*errResult = TrackedCString(err.Error())
```

Comment:
These comments are less of a review and more for my own education, but how does this error actually get passed to the C code and raised in python? It looks like the error is set here: `obfQuery = cb_obfuscate_sql(rawQuery, &error_message);` but there's something magic I'm missing, can you explain?

Reply:
This error string gets written into `error_message`, then we from c code we tell python to throw a runtime error containing that message using `PyErr_SetString(PyExc_RuntimeError, error_message);` ([more info here](https://docs.python.org/3/c-api/exceptions.html#c.PyErr_SetString))

Reply:
🤦 I completely missed the `errResult` in the signature here. No magic, this makes sense.